src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_MIN_CAPACITY = 10;
+    private Object[] array = new Object[ARRAY_MIN_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == array.length || size + 1 == array.length) {
+            array = grow();
+            copyArray(index);
+        } else if (index < size) {
+            copyArray(index);
+        }
```

Comment:
```suggestion
        } 
        copyArray(index);
```
код в двох умовах дублюєтся